File: cotgenerator/src/main/java/com/jon/cot/generator/utils/FileUtils.java
package com.jon.cot.generator.utils;

import androidx.annotation.RawRes;

import com.jon.cot.generator.CotApplication;

import org.apache.commons.io.IOUtils;

import java.io.FileInputStream;
import java.io.IOException;
import java.io.InputStream;

import timber.log.Timber;

public class FileUtils {
    private FileUtils() { /* blank */ }

    public static byte[] toByteArray(String filepath) throws IOException {
        return IOUtils.toByteArray(new FileInputStream(filepath));
    }

    public static byte[] toByteArraySafe(@RawRes int resId) {
        try {
            InputStream inputStream = CotApplication.getContext().getResources().openRawResource(resId);
            return IOUtils.toByteArray(inputStream);
        } catch (IOException e) {
            Timber.e(e);
            Timber.e("Error loading raw resource, this shouldn't happen!");
            return null;
        }
    }
}

File: common/src/main/java/com/jon/common/utils/FileUtils.kt
package com.jon.common.utils

import android.content.Context
import androidx.annotation.RawRes
import java.io.File

object FileUtils {
    fun toByteArray(filepath: String): ByteArray {
        return File(filepath).readBytes()
    }

    fun rawResourceToByteArray(context: Context, @RawRes resId: Int): ByteArray {
        val inputStream = context.resources.openRawResource(resId)
        return inputStream.readBytes()
    }
}
